Decouple `EditorInterface` from `EditorPlugin`

YuriSizov opened the pull request:
This is a part of https://github.com/godotengine/godot/pull/75694 that contains only the refactoring of `EditorInterface` into its own files. It also contains changes to includes, the way the singleton is accessed throughout the codebase, and exposes `EditorInterface.movie_maker_enabled` as a property.

In this PR the base class of the `EditorInterface` is still changed to `Object`, as there is no real reason for it to be a node. This should not be a problem for anyone using this singleton though.

YuriSizov committed:
Decouple EditorInterface from EditorPlugin

- Simplify some includes in the process.
- Also exposes EditorInterface.movie_maker_enabled as a property.

KoBeWi commented on editor/editor_interface.cpp:
```diff
@@ -0,0 +1,438 @@
+}
+
+void EditorInterface::create() {
+	ERR_FAIL_COND(singleton != nullptr);
```
This is redundant, because it's already checked in the constructor.

KoBeWi commented on editor/editor_interface.cpp:
```diff
@@ -0,0 +1,438 @@
+
+void EditorInterface::create() {
+	ERR_FAIL_COND(singleton != nullptr);
+	memnew(EditorInterface());
```
```suggestion
	memnew(EditorInterface);
```
This style is used commonly for `memnew()`.

YuriSizov replied:
Heh, yeah. I actually copied it from `EditorPaths` which I referenced here, and it also has this issue.

KoBeWi commented on editor/register_editor_types.cpp:
```diff
@@ -230,10 +231,16 @@ void register_editor_types() {
 void unregister_editor_types() {
 	EditorNode::cleanup();
+
+	if (EditorInterface::get_singleton()) {
```
Another redundant check, I think. `free()` method already checks if singleton exists. With an error, but this method shouldn't be called before calling `register_editor_types()`, so it would be wrong anyway.

YuriSizov commented:
Addressed the comments about checks, for `EditorPaths` as well. I briefly considered assigning the `singleton` value in the static method instead of the constructor, but ultimately decided that it makes sense to disallow any out-of-normal-flow construction and throw.

I kept the singleton check for `EditorPaths` on exist, though, as it is constructed manually in `main.cpp`, and thus can be skipped.

akien-mga commented:
Thanks!
